lib/dams/controller_helper.rb

```diff
@@ -914,12 +914,33 @@ def metadata_display?(data)
     #---
 
     def can_download?(document)
-      local_license = false
-      local_other_rights = document['otherRights_tesim'] && document['otherRights_tesim'].first.to_s.include?('localDisplay') ? true : false
-      if document['resource_type_tesim'] && document['resource_type_tesim'].first.to_s.include?('video')
-        local_license = document['license_tesim'] && document['license_tesim'].first.to_s.include?('localDisplay') ? true : false
+      license = document['license_tesim']
+      other_rights = document['otherRights_tesim']
+      return false if streaming_media_type?(document['resource_type_tesim']) && (local_display?(license) || local_display?(other_rights))
+      !metadata_only_display?(license) || !metadata_only_display?(other_rights)
+    end
+
+    def streaming_media_type?(resource_types)
+      resource_types.each do |t|
```

Comment:
For these support methods you might be able to use the `.any?` Enumerable method to avoid the explicit returns. Something like:

```
def streaming_media_type?(resource_types)
  resource_types.any? { |t| t.include?('video') || t.include?('sound') }
end
```